src/main/java/core/mate/academy/service/ExcavatorProducer.java

```diff
@@ -0,0 +1,15 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Excavator;
+import java.util.Arrays;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
```

Comment:
The `MachineProducer` interface should be parameterized to prevent non-Machine types from being used. Currently, it is correctly parameterized with `Excavator`, which is a subtype of `Machine`, so this adheres to the checklist requirement.